client/pluginManger.go

```diff
@@ -133,6 +139,26 @@ func (p *PluginManager) InstallPlugin(names []string) (err error) {
 	return
 }
 
+// InstallPluginWithVersion install a plugin by name & version
+func (p *PluginManager) installPluginWithVersion(name string) (err error) {
+	pluginAPI := PluginAPI{}
+	pluginName := "%s.hpi"
+	pluginVersion := strings.Split(name, "@")
+	url := fmt.Sprintf("http://updates.jenkins-ci.org/download/plugins/%s/%s/%s.hpi",
+		pluginVersion[0], pluginVersion[1], pluginVersion[0])
+
+	statusCode := pluginAPI.download(url, name)
+
+	if statusCode >= 200 && statusCode < 300 {
+		if err = p.Upload(fmt.Sprintf(pluginName, name)); err != nil {
+			return err
```

Comment:
The plugin file will not deletes if there's anything goes wrong here. Usually, we download a temporary file into a temporary directory.